test/torchaudio_unittest/sox_io_backend/save_test.py

```diff
@@ -415,10 +416,25 @@ def test_tensor_preserve(self, dtype):
         expected = get_wav_data(dtype, 4)[::2, ::2]
 
         data = expected.clone()
-        sox_io_backend.save(path, data, 8000)
+        sox_io_backend.save(path, data, 8000, dtype=None)
 
         self.assertEqual(data, expected)
 
+    @parameterized.expand([
+        ('float32', (-1, 1)),
+        ('int32', (-2147483648, 2147483647)),
+        ('int16', (-32768, 32767)),
+        ('uint8', (0, 255)),
+    ])
+    def test_dtype_conversion(self, dtype, range):
+        """`save` performs dtype conversion on float32 src tensors only."""
+        path = self.get_temp_path("data.wav")
+        data = get_wav_data("float32", 2)
+        sox_io_backend.save(path, data, 8000, dtype=dtype)
+        found = load_wav(path, normalize=False)[0]
+        self.assertEqual(found.dtype, getattr(torch, dtype))
+        assert(torch.all(found >= range[0]) and torch.all(found <= range[1]))
```

Comment:
This range assertion does not look quite right. In case of `int32` conversion, it will still pass even if the conversion does not happen.

How about performing exact comparison with Tensors that are small enough to catch the element wise parity?

```python
    @parameterized.expand([
         ('float32', torch.Tensor([-1.0, -0.5, 0, 0.5, 1.0]).to(torch.float32)),
         ('int32', torch.Tensor([-2147483648, ..., 0, ..., 2147483647]).to(torch.int32)),
        ...
    ])
    def test_dtype_conversion(self, dtype, expected):
        data = torch.Tensor([-1.0, -0.5, 0, 0.5, 1.0]).to(torch.dtype32)
        ...
        self.assertEqual(found, expected)
```